pkg/kubelet/volume/secret/secret.go

```diff
@@ -97,7 +97,7 @@ func (sv *secretVolume) SetUp() error {
 		return fmt.Errorf("Cannot setup secret volume %v because kube client is not configured", sv)
 	}
 
-	secret, err := kubeClient.Secrets(sv.secretRef.Namespace).Get(sv.secretRef.Name)
+	secret, err := kubeClient.Secrets(sv.podRef.Namespace).Get(sv.secretRef.Name)
```

Comment:
I'm not convinced this is clearer.  We still need secretRef anyway, so it feels more self-documenting to just leave this as it was, unless there's a plan to kill off secretRef